src/cirq_iqm/iqm_sampler.py

```diff
@@ -149,3 +166,13 @@ def _send_circuits(  # pylint: disable=too-many-arguments
             raise RuntimeError('No measurements returned from IQM quantum computer.')
 
         return [{k: np.array(v) for k, v in measurements.items()} for measurements in results.measurements]
+
+    def _metadata_from_architecture(self):
+        arch = self._client.get_quantum_architecture()
+        qubits = frozenset(cirq.NamedQubit(qb) for qb in arch.qubits)
+        connectivity = (
+            {int(qb.removeprefix(IQMDeviceMetadata.QUBIT_NAME_PREFIX)) for qb in edge}
+            for edge in arch.qubit_connectivity
+        )
+        gateset = cirq.Gateset(*(_IQM_CIRQ_OP_MAP[op] for op in arch.operations if _IQM_CIRQ_OP_MAP[op] is not None))
+        return IQMDeviceMetadata(qubits, connectivity, gateset)
```

Comment:
I would prefer to have that as a method of `IQMDeviceMetadata` taking the `quantum_architecture` as an argument.

Reply:
done